ceph/mds.go

```diff
@@ -0,0 +1,173 @@
+package ceph
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	cephCmd                      = "/usr/bin/ceph"
+	mdsBackgroundCollectInterval = 5 * time.Minute
+)
+
+const (
+	MDSModeDisabled   = 0
+	MDSModeForeground = 1
+	MDSModeBackground = 2
+)
+
+type mdsStat struct {
+	FSMap struct {
+		Filesystems []struct {
+			MDSMap struct {
+				FSName string `json:"fs_name"`
+				Info   map[string]struct {
```

Comment:
next PR should capture more of the mdsmap fields; to compare MDS against:
- epoch
- max_mds
- in/failed/damaged/stopped/up

Reply:
Agreed! This is just a PoC to get our foot in the door and the get metrics scraping right. Once we have MDSs being scraped world is our oyster. :)